### DISCUSSION TITLE
请教下如何监听kook的按钮事件

### DISCUSSION THREAD

**Discussion by Crazymiku** [simple-robot/Discussions#193] [Q&A] [answered]
### 涉及的编程语言

Java

### 涉及的组件库

KOOK

### 疑问描述

我使用下面的代码无法捕捉到按钮事件，按钮定义了click：return-val，返回值value：test
通过在KookBotEventsKt的断点确定是已接收到点击事件，但估计哪里值没匹配对导致下面代码无法生效。
使用的包为
simbot-core-spring-boot-starter:4.5.0
simbot-component-kook-core:4.0.0-beta5
`@Listener
  @Filter(value = "test", matchType = MatchType.TEXT_CONTAINS)
  public void test(KookMessageBtnClickEvent event) {}`

**Comment by ForteScarlet** [comment, marked as answer]
我记得 `KookMessageBtnClickEvent` 并非 **消息事件**，试着去掉注解 `@Filter` 看看？
`@Filter` 的文本内容匹配仅对存在文本内容的消息事件生效（默认情况下）
